runtime/src/main/java/org/corfudb/protocols/wireprotocol/StreamAddressSpace.java

```diff
@@ -0,0 +1,140 @@
+package org.corfudb.protocols.wireprotocol;
+
+import org.roaringbitmap.longlong.LongIterator;
+import org.roaringbitmap.longlong.Roaring64NavigableMap;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.NavigableSet;
+import java.util.concurrent.atomic.AtomicLong;
+
+/**
+ * This class represents the space of addresses of a stream.
+ *
+ * A stream's address space is defined by:
+ *       1. The collection of all addresses that belong to this stream.
+ *       2. The trim mark (last trimmed address, i.e., an address that is no longer present and that was subsumed by
+ *       a checkpoint).
+  *
+ * Created by annym on 03/06/2019
+ */
+public class StreamAddressSpace {
+
+    // Holds the last trimmed address for this stream.
+    // Note: keeping the last trimmed address is required in order to properly set the stream tail on sequencer resets
+    // when a stream has been checkpointed and trimmed and there are no further updates to this stream.
+    private AtomicLong trimMark;
+
+    // Holds the complete map of addresses for this stream (bitmap).
+    private Roaring64NavigableMap addressMap;
+
+    public StreamAddressSpace(long trimMark, Roaring64NavigableMap addressMap) {
+        this.trimMark = new AtomicLong(trimMark);
+        this.addressMap = addressMap;
+    }
+
+    public synchronized Roaring64NavigableMap getAddressMap() {
+        return addressMap;
```

Comment:
why synchronized? Is Roaring64NavigableMap not threadsafe?

Reply:
It is not thread-safe.